UI/backend/app.py

```diff
@@ -11,106 +15,110 @@
+    except Exception as db_error:
+        return jsonify({'success': False, 'message': str(db_error)}), requests.codes.internal_server_error
 
 @app.route('/addTimeslot', methods=['POST'])
 def add_time_slot():
     data = request.json
-    ts = data.get('timeSlot', '')
-
+    ts = data.get(UniJsonKeys.TIME_SLOT, '')
     if not ts:
-        return jsonify({'success': False, 'message': 'Time slot is required'}), 400
+        return jsonify({'success': False, 'message': 'Time slot is required'}), requests.codes.bad_request
     
-    print(f"Received data: {data}")
-
-    app_dir = os.path.dirname(os.path.abspath(__file__))
-    grand_parent_dir = os.path.dirname(os.path.dirname(app_dir))
-    cpp_exec_path = os.path.join(grand_parent_dir, 'run.sh')
-
+    ts_members = ts.split()
+    if len(ts_members) != 3:
+        return jsonify({'success': False, 'message': 'Invalid time slot format'}), requests.codes.bad_request
+    day, start_time, end_time = ts_members
     try:
-        result = subprocess.run(['bash', cpp_exec_path, '--addTimeslot', ts], capture_output=True, text=True)
-
-        if result.returncode == 0:
-            return jsonify({'success': True, 'message': 'Time slot added successfully!'})
-        else:
-            return jsonify({'success': False, 'message': 'Failed to add time slot', 'details': result.stderr}), 500
-    except Exception as e:
-        print(f"Exception occurred: {str(e)}")
-        return jsonify({'success': False, 'message': str(e)}), 500
+        insert_time_slot(day, start_time, end_time)
+        return jsonify({'success': True, 'message': 'Time slot added successfully!'})
+    except Exception as db_error:
+        return jsonify({'success': False, 'message': str(db_error)}), requests.codes.internal_server_error
     
 @app.route('/addInstructor', methods=['POST'])
 def add_instructor():
     data = request.json
-    inst = data.get('instructor', '')
-    course = data.get('courseNameInst', '')
-    time  = data.get('timeSlotInst', '')
-
-
+    inst = data.get(UniJsonKeys.INSTRUCTOR, '')
+    course = data.get(UniJsonKeys.COURSE_NAME_INST, '')
+    time  = data.get(UniJsonKeys.TIME_SLOT_INST, '')
     if not inst:
-        return jsonify({'success': False, 'message': 'Instructor is required'}), 400
-    print(f"Received data: {data}")
-
-    app_dir = os.path.dirname(os.path.abspath(__file__))
-    grand_parent_dir = os.path.dirname(os.path.dirname(app_dir))
-    cpp_exec_path = os.path.join(grand_parent_dir, 'run.sh')
-
+        return jsonify({'success': False, 'message': 'Instructor is required'}), requests.codes.bad_request
+    insert_func_args = [inst]
+    if course:
+        insert_func_args.append(course)
+        insert_course_args = [course]
+    if not course and time:
+        return jsonify({'success': False, 'message': 'If time is inputed course is mandatory'}), requests.codes.bad_request
+    if time:
+        ts_members = time.split()
```

Comment:
you have a similar code part in 'add_course'. Move them a function and call it in both places

Reply:
done